lib/ruby_lsp/listeners/code_lens.rb

```diff
@@ -42,6 +42,7 @@ def initialize(response_builder, global_state, uri, dispatcher)
         @group_stack = T.let([], T::Array[String])
         @group_id = T.let(1, Integer)
         @group_id_stack = T.let([], T::Array[Integer])
+        @def_depth = T.let(0, Integer)
```

Comment:
For the benefit of future maintainers, I think this is worth a brief explanation:
```suggestion
        # We want to avoid adding code lenses for nested definitions
        @def_depth = T.let(0, Integer)
```